test/xpu_api/tuple/tuple.tuple/tuple.elem/cv_tuple_element_g.pass.cpp

```diff
@@ -0,0 +1,69 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/tuple>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+#include "support/utils_invoke.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+void
+kernel_test1(sycl::queue& deviceQueue)
+{
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        cgh.single_task<class KernelTest1>([=]() {
+            {
+                static_assert(
+                    dpl::is_same<dpl::tuple_element<0, const dpl::tuple<float, void, int>>::type, const float>::value);
+                static_assert(
+                    dpl::is_same<dpl::tuple_element<1, volatile dpl::tuple<short, void>>::type, volatile void>::value);
+                static_assert(dpl::is_same<dpl::tuple_element<2, const volatile dpl::tuple<float, char, int>>::type,
+                                           const volatile int>::value);
```

Comment:
Can we improve readability and improve coverage here:

```suggestion
                using tuple_type = dpl::tuple<float, short, int>;
                static_assert(dpl::is_same_v<dpl::tuple_element<0, const tuple_type>::type, const float>);
                static_assert(dpl::is_same_v<dpl::tuple_element<1, volatile tuple_type>::type, volatile short>);
                static_assert(dpl::is_same_v<dpl::tuple_element<2, const volatile tuple_type>::type, const volatile int>);
                
                static_assert(dpl::is_same_v<dpl::tuple_element_t<0, const tuple_type>, const float>;
                static_assert(dpl::is_same_v<dpl::tuple_element_t<1, volatile tuple_type>, volatile short>);
                static_assert(dpl::is_same_v<dpl::tuple_element_t<2, const volatile tuple_type>, const volatile int>);
```